pdr_backend/contract/feed_contract.py

```diff
@@ -25,9 +25,20 @@ def __init__(self, web3_pp, address: str):
         # return Wei(0) for unknown keys
         self.last_allowance: Dict[str, Wei] = defaultdict(lambda: Wei(0))
 
+        # cache token symbol for logging
+        self._token_symbol: Optional[str] = None
+
     def set_token(self, web3_pp):
-        stake_token = self.get_stake_token()
-        self.token = Token(web3_pp, stake_token)
+        USDC = self.get_USDC()
```

Comment:
As above, if the smart contract already has `get_stake_token()` then let's leave it as that. Minimize rocking the boat.